server/server.go

```diff
@@ -90,35 +97,79 @@ func (s *RpcEndPointServer) Start() {
 		}
 	}()
 
-	// Handler for root URL (JSON-RPC on POST, public/index.html on GET)
-	http.HandleFunc("/", s.HandleHttpRequest)
-	http.HandleFunc("/health", s.handleHealthRequest)
-	http.HandleFunc("/bundle", s.HandleBundleRequest)
+	s.startMainServer()
+	s.startDrainServer()
+
+	notifier := make(chan os.Signal, 1)
+	signal.Notify(notifier, os.Interrupt, syscall.SIGTERM)
+
+	<-notifier
```

Comment:
My understanding is that we don't need to have separate server for that and call "drain" method.

All we need is to: 
1. start returning errors from /health after this line is crossed
2. Wait n-seconds here instead of in the drain handler

If its the same then I would prefer to do it here with one server because its more simple and straightforward. 

But maybe its not the same?

Reply:
I also thought about that idea (sleep in the `SIGTERM` handler).

reason I did not go for it was that we only need to do that (sleeping) when we run it in k8s behind AWS LB target group (because it takes its time to become aware of the pod being drained).  while in the case of plain k8s service, the service becomes aware of the pod being drained basically immediately (I guess the control plane first updates the service's endpoints, and only after sends a signal to the pod).

if we introduce the sleep into SIGTERM handler then we'd impose that on all workflows regardless if that would be necessary or not.

Reply:
I mean something like this https://github.com/flashbots/rpc-endpoint/pull/116

Reply:
You probably right, I agree. Did not see your comment